### PR TITLE
KT-8286; part `String.toInt()`

### PR DESCRIPTION
**Pull request by voddan**

### CODE REVIEW

**Comment by yingzhuo** [comment]
Nice!  `String.toXxxOrNull` is very handy.

File: libraries/stdlib/src/kotlin/text/ParsePrimitivesJVM.kt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+ */
+@kotlin.internal.InlineOnly
+public inline fun String.toLong(radix: Int = 10): Long = java.lang.Long.parseLong(this, radix)
+
+/**
+ * Parses the string as a [Float] number and returns the result.
+ * @throws NumberFormatException if the string is not a valid representation of a number.
+ */
+@kotlin.internal.InlineOnly
+public inline fun String.toFloat(radix: Int = 10): Float = java.lang.Float.parseFloat(this)
```
Comment (HaasJona) [inline libraries/stdlib/src/kotlin/text/ParsePrimitivesJVM.kt]: Radix parameter seems to be unused

File: libraries/stdlib/src/kotlin/text/ParsePrimitivesJVM.kt [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+/**
+ * Parses the string as a [Double] number and returns the result.
+ * @throws NumberFormatException if the string is not a valid representation of a number.
+ */
+@kotlin.internal.InlineOnly
+public inline fun String.toDouble(radix: Int = 10): Double = java.lang.Double.parseDouble(this)
```
Comment (HaasJona) [inline libraries/stdlib/src/kotlin/text/ParsePrimitivesJVM.kt]: Radix parameter seems to be unused

File: libraries/stdlib/src/kotlin/text/ParsePrimitivesJVM.kt (hunk 1 above)
Comment (voddan) [inline libraries/stdlib/src/kotlin/text/ParsePrimitivesJVM.kt]: will fix, thank you

**Comment by ilya-g** [comment]
I've squashed several commits, made some improvements and merged it manually.
Thanks for the great work.

**Pull request closed without merging**